### DISCUSSION TITLE
Where do I configure resources for strimzi cluster operator zookeeper pods using kubectl

### DISCUSSION THREAD

**Discussion by marblerun** [strimzi/strimzi-kafka-operator#10218] [Q&A]
Morning,

Simple question. After a heavy production run was scheduled, both our clickhouse and kafka ZK clusters became somewhat overwhelmed, and entered a continuous cycle of crashloopbackoff, due to the failure of healthchecks. As the clickhouse cluster has a statefulset definition, I've been able to increase cpu resource by patching the statefulset, and some sort of stability has returned.

Checking the pod logs showed there was an issue talking to kafka, and it appears its ZK component has similar issue. I hoped to be able to just repeat the patch procedure, but unless I'm mistaken, there is no statefuleset to edit.

I believe our kafka cluster was created via applying an appropriate yaml file with the strimzi operater running. It has a small zookeeper section covering the number of replicas, logging, metricsConfig and an edited storage section pointing to a 5GB pvc. 

Is this where I need to add a resource section, or can it be applied as a seperate yaml file ? All I need is a quick explanation, a basic example file and the where to apply, if it isn't just kubectl apply -f zk-kafka-resource.yaml, for instance.

kind regards,

Mike

**Comment by scholzj** [comment]
When using an operator, you should never patch any of the resources managed by the operator. You should also configure the things such as resources in the custom resources provided by the operator. For Strimzi based Kafka and ZooKeeper, you can do it in the `Kafka` custom resource. Please check the docs for the exact placement and examples.

**Comment by marblerun** [comment]
Morning,

if the docs made this clear, I probably wouldn't be asking. 

https://strimzi.io/docs/operators/in-development/deploying#config-resources-str

By default, the Strimzi Cluster Operator does not specify CPU and memory resource requests and limits for its deployed operands. Ensuring an adequate allocation of resources is crucial for maintaining stability and achieving optimal performance in Kafka. The ideal resource allocation depends on your specific requirements and use cases.

It is recommended to configure CPU and memory resources for each container by [setting appropriate requests and limits](https://strimzi.io/docs/operators/in-development/configuring#con-common-configuration-resources-reference).

so skip to that section, where we find a reasonable resource example in section 2.5

My question remains, but it should perhaps be, "how do I edit existing resources using kubectl ? " Just edit down my existing config, add a resource section for cpu and memory, and apply ?

Thanks

**Reply by scholzj** [reply]
You can edit the custom resource as any other resource. E.g. as the StatefulSet you wanted to edit before. So for example with `kubectl edit kafka <resourcename> -n <namespacename>`. Or if you prefer to use YAML files you can edit it in a YAML file and just do `kubectl apply -f ...`. The custom resource is used as any other Kubernetes resource. 

Our docs expect you to know how to use Kubernetes. We cannot really document in detail how to use Kubernetes APIs or tools such as `kubectl`. We do not have the resources for that and it is anyway covered by Kubernetes documentation.
